### PR TITLE
CONN-2238 loadtestdata

### PR DESCRIPTION
**Pull request by trantang**
moving the loadtestdata to webservice

### CODE REVIEW

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/LoadTestDataDocumentBean.java [hunk 1]
```diff
@@ -70,56 +74,64 @@
...
     private Map<String, String> listStatusType;
+    private List<Document> documentList;
+    Map<Long, Patient> lookupPatient;
 
     public LoadTestDataDocumentBean() {
         selectedDocument = null;
     }
 
     // selected-record
-    public DocumentMetadata getSelectedDocument() {
```
Comment (plobre) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/LoadTestDataDocumentBean.java:88]: Curious, why was this switched to "Document"? With the new Document Data Submission service, we can upload a document meta data without uploading a document - this leads me to believe we may introduce errors in this case.

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/configuration/GenericPortDescriptor.java [hunk 2]
```diff
@@ -24,38 +24,31 @@
...
-import javax.xml.ws.soap.SOAPBinding;
+import gov.hhs.fha.nhinc.messaging.service.port.SOAP12ServicePortDescriptor;
 
 /**
+ * @author ttang
  *
- * @author Neil Webb
  */
-@BindingType(value = SOAPBinding.SOAP12HTTP_BINDING)
-public class PolicyEngineTransformationService implements
```
Comment (plobre) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/configuration/GenericPortDescriptor.java:38]: @trantang  What is going on here? Why are we renaming this file, and completely changing its implementation?

**Review by plobre — CHANGES_REQUESTED** [review]

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/configuration/GenericPortDescriptor.java (hunk 2 above)
Comment (trantang) [reply Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/configuration/GenericPortDescriptor.java:38]: this is a new-file not a rename i don't know why it changing the `PolicyEngineTransformationService`

File: Product/Production/Gateway/AdminGUIWebservices/src/main/java/gov/hhs/fha/nhinc/admingui/loadtestdata/LoadTestDataManagement.java [hunk 3]
```diff
@@ -0,0 +1,507 @@
...
+    private static final String ACT_SUCCESSFUL = "successful";
+    private static final String ACT_FAIL = "fail";
+
+    @Autowired
+    private LoadTestDataService loadTestDataService;
+
+    @Override
+    public LoadTestDataSimpleResponseMessageType deleteAddress(DeleteAddressRequestMessageType request) {
+        Address record = new Address(request.getAddress(), loadTestDataService.getPatientBy(request.getAddress().getPatientId()));
+        if (null == record) {
```
Comment (plobre) [inline Product/Production/Gateway/AdminGUIWebservices/src/main/java/gov/hhs/fha/nhinc/admingui/loadtestdata/LoadTestDataManagement.java]: What are all these null checks for? These can never be null - You're creating the object in the lines directly above?

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/LoadTestDataDocumentBean.java (hunk 1 above)
Comment (trantang) [reply Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/LoadTestDataDocumentBean.java:88]: the document is intended to handle some of the issue that came up during the adminGUI interacting with the model; xml-date > Date > string -- issue with the converter and additional properties that is only used by the adminGUI

File: Product/Production/Gateway/AdminGUIWebservices/src/main/java/gov/hhs/fha/nhinc/admingui/loadtestdata/LoadTestDataManagement.java [hunk 4]
```diff
@@ -0,0 +1,507 @@
...
+            return buildSimpleResponse(Boolean.FALSE, ACT_FAIL);
+        }
+        loadTestDataService.deleteAddress(record);
```
Comment (plobre) [inline Product/Production/Gateway/AdminGUIWebservices/src/main/java/gov/hhs/fha/nhinc/admingui/loadtestdata/LoadTestDataManagement.java]: Why are we assuming this passes? This method returns a boolean if it is successful or not - use that to build an ACT_SUCCESSFUL or ACT_FAIL message. As of right now its always a Success

This issue is repeated for most of these services

File: Product/Production/Gateway/AdminGUIWebservices/src/main/java/gov/hhs/fha/nhinc/admingui/loadtestdata/LoadTestDataManagement.java [hunk 5]
```diff
@@ -0,0 +1,507 @@
...
+        return buildSimpleResponse(Boolean.FALSE, ACT_FAIL);
+    }
+
+    @Override
+    public LoadTestDataSimpleResponseMessageType saveEventCode(SaveEventCodeRequestMessageType request) {
+        DocumentMetadata meta = loadTestDataService.getDocumentBy(request.getEventCode().getDocumentid());
+        if (null != meta) {
+            try {
+                EventCode record = new EventCode(request.getEventCode(), meta);
+                loadTestDataService.saveEventCode(record);
```
Comment (plobre) [inline Product/Production/Gateway/AdminGUIWebservices/src/main/java/gov/hhs/fha/nhinc/admingui/loadtestdata/LoadTestDataManagement.java:415]: Should use the result to determine if it passed or failed - Same issue applies for most of these

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/configuration/GenericPortDescriptor.java (hunk 2 above)
Comment (trantang) [reply Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/configuration/GenericPortDescriptor.java:38]: this file got deleted and new-file got add that why it see this as a rename/move and re-implemented.

**Review by plobre — CHANGES_REQUESTED** [review]

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/LoadTestDataDocumentBean.java [hunk 6]
```diff
@@ -70,56 +74,64 @@
...
 
     // database-methods
-    public List<DocumentMetadata> getDocuments() {
-        List<DocumentMetadata> documents = loadTestDataService.getAllDocuments();
-        Map<Long, Patient> lookupPatient = mapPatientById(loadTestDataService.getAllPatients());
-        for (DocumentMetadata doc : documents) {
-            if (doc.getPatientRecordId() != null) {
-                doc.setPatientIdentifier(lookupPatient.get(doc.getPatientRecordId()).getPatientIdentifier());
+    public List<Document> getDocuments() {
+        if (isCollectionEmpty(lookupPatient)) {
```
Comment (TabassumJafri) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/LoadTestDataDocumentBean.java]: Can we use CollectionUtils.isEmpty here?

**Review by plobre — APPROVED** [review]

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/LoadTestDataDocumentBean.java [hunk 7]
```diff
@@ -291,24 +319,16 @@ public void setPatientId(String patientIdentifierIso) {
 
         if (StringUtils.isNotBlank(patientIdentifierIso)) {
             String[] identifierValue = patientIdentifierIso.split("&");
-            HelperUtil.updateDocumentBy(withDocument,
-                loadTestDataService.getPatientBy(identifierValue[0].replace("^^^", ""), identifierValue[1]));
+            CoreHelpUtils.updateDocumentBy(withDocument,
+                wsService.getPatientBy(identifierValue[0].replace("^^^", ""), identifierValue[1]));
         } else {
-            HelperUtil.updateDocumentBy(withDocument, new Patient());
+            throw new RuntimeException("PatientIdentifier should not be empty.");
```
Comment (TabassumJafri) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/LoadTestDataDocumentBean.java]: Need to throw a customized exception

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/model/loadtestdata/Document.java [hunk 8]
```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2009-2018, United States Government, as represented by the Secretary of Health and Human Services.
```
Comment (TabassumJafri) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/model/loadtestdata/Document.java]: Fix the copyright here

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/model/loadtestdata/Patient.java [hunk 9]
```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2009-2018, United States Government, as represented by the Secretary of Health and Human Services.
```
Comment (TabassumJafri) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/model/loadtestdata/Patient.java]: Fix copyright

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/services/LoadTestDataWSService.java [hunk 10]
```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2009-2018, United States Government, as represented by the Secretary of Health and Human Services.
```
Comment (TabassumJafri) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/services/LoadTestDataWSService.java]: Fix copyright

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/services/impl/LoadTestDataWSServiceImpl.java [hunk 11]
```diff
@@ -0,0 +1,656 @@
+/*
+ * Copyright (c) 2009-2018, United States Government, as represented by the Secretary of Health and Human Services.
```
Comment (TabassumJafri) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/services/impl/LoadTestDataWSServiceImpl.java]: Fix copyright

File: Product/Production/Gateway/AdminGUIWebservices/src/main/java/gov/hhs/fha/nhinc/admingui/loadtestdata/LoadTestDataManagement.java [hunk 12]
```diff
@@ -0,0 +1,507 @@
+/*
+ * Copyright (c) 2009-2018, United States Government, as represented by the Secretary of Health and Human Services.
```
Comment (TabassumJafri) [inline Product/Production/Gateway/AdminGUIWebservices/src/main/java/gov/hhs/fha/nhinc/admingui/loadtestdata/LoadTestDataManagement.java]: fix copyright

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/LoadTestDataPatientBean.java [hunk 13]
```diff
@@ -223,17 +237,22 @@ public void newAddress() {
...
-            if (CollectionUtils.isNotEmpty(identifierList) && identifierList.size() > 1) {
-                result = loadTestDataService.deleteIdentifier(selectedIdentifier);
+            if (CollectionUtils.isNotEmpty(getIdentifiers()) && getIdentifiers().size() > 1) {
+                result = wsService.deleteIdentifier(selectedIdentifier);
+                if (result) {
+                    withPatient.getIdentifierList().remove(selectedIdentifier);
+                }
                 selectedIdentifier = null;
             } else {
                 addPatientErrorMessages("Patient-identifier cannot be empty: fail to delete the last record");
```
Comment (TabassumJafri) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/LoadTestDataPatientBean.java]: Update message as to say Last identifier cannot be deleted

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/nhinclib/NhincConstants.java [hunk 14]
```diff
@@ -698,6 +698,40 @@ public static EVENT_LOGGING_SERVICE_NAME fromValueString(String valueString) {
     public static final String ADMIN_DASHBOARD_ERRORLOG_GETFILTERS = "getSearchFilter";
     public static final String ADMIN_DASHBOARD_ERRORLOG_VIEW = "viewErrorLog";
 
+    // load-test-data: ADMIN_LTD_SAVE/LIST/DELETE/GET
+    public static final String ENTITY_LOAD_TEST_DATA_SERVICE_NAME = "entityloadtestdata";
+    public static final String ADMIN_LTD_DELETEADDRESS = "deleteAddress";
+    public static final String ADMIN_LTD_DELETEIDENTIFIER = "deleteIdentifier";
```
Comment (TabassumJafri) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/nhinclib/NhincConstants.java]: Can we move these constants into AdminGUI/AdminGUIWebservices project ..which ever project is more appropriate.

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/util/CoreHelpUtils.java [hunk 15]
```diff
@@ -118,4 +131,110 @@ public static Date getDate(String dateString) {
...
+
+    public static <R, T> List<R> castListType(List<T> patientList) {
+        List<R> list = new ArrayList<>();
+        for (T patient : patientList) {
+            list.add((R) patient);
+        }
+        return list;
+    }
+
+    public static <K, V> boolean isCollectionEmpty(Map<K, V> map) {
```
Comment (TabassumJafri) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/util/CoreHelpUtils.java:232]: Why we are recreating CollectionUtils.isEmpty()?

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/LoadTestDataDocumentBean.java [hunk 16]
```diff
@@ -144,7 +161,10 @@ public void duplicateDocument() {
     public boolean deleteDocument() {
         boolean result = false;
         if (selectedDocument != null) {
-            result = loadTestDataService.deleteDocument(selectedDocument);
+            result = wsService.deleteDocument(selectedDocument);
+            if(result){
+                getDocuments().remove(selectedDocument);
```
Comment (TabassumJafri) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/LoadTestDataDocumentBean.java]: Personal Preference: do not like removing from the list. Can we fetch here?

**Review by TabassumJafri — CHANGES_REQUESTED** [review]

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/util/CoreHelpUtils.java (hunk 15 above)
Comment (trantang) [reply Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/util/CoreHelpUtils.java:232]: going to use maputils

**Review by TabassumJafri — APPROVED** [review]

**Review by ptambellini01 — APPROVED** [review]

**Pull request merged**